zeyple/zeyple.py

```diff
@@ -83,6 +89,78 @@ def gpg(self):
 
         return ctx
 
+    def _validate_dnssec(self, domain):
+        # if force_dnssec is enabled go ahead
+        if self.config.has_option('zeyple', 'force_dnssec') and \
+            self.config.getboolean('zeyple', 'force_dnssec'):
+
+            dnssec_check = urllib.urlopen("http://portfolio.sidnlabs.nl:8080/check/" + domain).read().split(',')
+            if dnssec_check[2] == "secure":
+                logging.info("DNSSec: " + domain + " is secure")
+                return True
+            else:
+                logging.warn("DNSSec: " + domain + " is insecure")
+                return False
+        else:
+            return True
+
+    def _retrieve_pka_key(self, recipient):
+        username,domainname = recipient.split("@")
```

Comment:
That is extremely fragile code.

*Did you know?* Email address can actually have multiple at signs. According to the [RFC 5322](http://tools.ietf.org/html/rfc5322) `"john@doe"@example.com" is a perfectly valid email address, it will send an email to `john@doe` on the server `example.com`.

Anyway, you should use `rsplit()` here.